Potentially Breaking Change: Update ADLA catalog

begoldsm opened the pull request:
<!-- DO NOT DELETE THIS TEMPLATE -->

## Description
This change updates the ADLA catalog to do the following:
1. Introduces the new package item type
2. Allows for listing the following items from their parent's parent item (the database):
  * Table
  * view
  * Table valued function
  * Table statistics
3. Switched the FileType enum to a string since this field is read only and will have new types added and changed with some regularity. Note that this is a "breaking" change but it is being made as part of the preview release which has not yet been announced/consumed. I updated the middle version to reflect this, but let me know if we require a major version update for this preview release and I will make it.
---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md).**
- [ ] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/Documentation/breaking-changes.md).**

### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#sdk-generation-guidelines)
- [x] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [x] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as a link to the swagger spec, used to generate the code.
- [x] The `project.json` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

Commit:
Update to latest swagger spec for ADL

This change affects catalog functionality:
* Add new package catalog item type
* Update to allow for listing of common catalog items from a base
database:
- Table
- TableValuedFunction
- TableStatistics
- Views

Updates for tests and versions.

comment out new tests for ADL

doing this until i can run tests in my local environment.

Update tests and recordings.

Actually copying the right file to the right place.

update changelog.

markcowl commented:
@begoldsm You may want to pull the latest from this PR: #3117 

Also, please fill out the checklist

begoldsm commented:
@markcowl shame on me, I know better! :). 
Here is the PR for the swagger update: https://github.com/Azure/azure-rest-api-specs/pull/1119

begoldsm commented:
@markcowl @shahabhijeet I have synced to the latest and squahsed my commits down and update the title, description and the checklist. Let me know if you have any questions or need anything else from me!

shahabhijeet commented on src/SDKs/DataLake.Analytics/Management.DataLake.Analytics/Microsoft.Azure.Management.DataLake.Analytics.csproj:
```diff
@@ -34,6 +34,7 @@
 
   <PropertyGroup>
     <TargetFrameworks>net452;netstandard1.4</TargetFrameworks>
+    <Version>2.2.0-preview</Version>
```
@begoldsm let's continue using <versionPrefix> as that is being used across, but if you want to use <version> then you don't need <versionPrefix>. Also you do not need any package related properties.
All you need is the PackageId, assemblyName, assembly description and package tags. All the remaining ones not needed.
Also remove all commented code.

shahabhijeet commented on src/SDKs/DataLake.Store/Management.DataLake.Store/generate.cmd:
```diff
@@ -6,8 +6,8 @@
 
-set accountSpecFile="https://raw.githubusercontent.com/Azure/azure-rest-api-specs/0ade6547eff89bab427eb400ebc5da870f2c5bd0/arm-datalake-store/account/2016-11-01/swagger/account.json"
-set filesystemSpecFile="https://raw.githubusercontent.com/Azure/azure-rest-api-specs/0ade6547eff89bab427eb400ebc5da870f2c5bd0/arm-datalake-store/filesystem/2016-11-01/swagger/filesystem.json"
+set accountSpecFile="https://raw.githubusercontent.com/Azure/azure-rest-api-specs/12a6c2299296b6877fa1e5a03ed42752445877dc/arm-datalake-store/account/2016-11-01/swagger/account.json"
```
@begoldsm was this missed as part of earlier checkin for the DataLake Store?

begoldsm replied:
no I just want to keep all specs at the same commit level.

shahabhijeet replied:
@begoldsm URL, licensense agreement, generating assembly version info. All this is getting controlled from common properties. So removing them will be really make all the nuget consistent

begoldsm replied:
@shahabhijeet this has been fixed and I removed all other entries. I am also, in the next iteration, moving the package ID to the top for easier identification.

Commit:
Cleanup ADL after migration to dev17

This cleans up files that are no longer used and updates csproj files to
remove commented out sections

Commit:
Added a number of null checks to ADLS client

We were getting some null reference exceptions on occasion so ensuring
none of these nested properties are null before indexing to them.

Commit:
Move package ID to the top to address PR.
